tests/introspection/tests.py

```diff
@@ -189,6 +184,15 @@ def test_get_indexes_multicol(self):
         self.assertNotIn('first_name', indexes)
         self.assertIn('id', indexes)
 
+    @skipUnlessDBFeature('can_introspect_index_type')
+    def test_get_constraints_indexes_types(self):
+        with connection.cursor() as cursor:
+            constraints = connection.introspection.get_constraints(cursor, Article._meta.db_table)
+        for key, val in constraints.items():
+            if val['index'] and not (val['primary_key'] or val['unique']):
```

Comment:
It can, in commits like this one - https://github.com/django/django/commit/324c1b432a2fa07c8f911f8b923ed04be32c52c7.
And then this test will also require modification (for no good).

Another thing that I can do is use `Index` instead of `index_together` here and set the index name explicitly (using `Index(name='foo_bar'`). What do you think? Maybe I am overthinking?